docs/querying/functions.md

```diff
@@ -50,6 +50,24 @@ For each input time series, `changes(v range-vector)` returns the number of
 times its value has changed within the provided time range as an instant
 vector.
 
+## `clamp()`
+
+`clamp(v instant-vector, min scaler, max scalar)` clamps the sample 
+values of all elements in `v` to have an upper limit of `max` and lower limit of `min`.
+
+For example if the `v` has the values [0, 1, 2, 3, 4]
+
+`clamp(v, 1.5, 3.5)` would give the result [1.5, 1.5, 2, 3, 3.5].
+
+`clamp(v instant-vector, min scaler, max scalar, val scaler)` replace the
+values in `v` having an upper limit of `max` and lower limit of `min` with `val`,
+and replaces the edges with `val/2`.  This is similar to the rectangular function.
+https://en.wikipedia.org/wiki/Rectangular_function
```

Comment:
It is not similar to the rectangle function, which has similar input. I don't think we need to have the edges replaced as well.

Reply:
I am back peddling here and would like to forego the idea of overloading the clamp function.  My worry here, after thinking about it, is that this would cause more confusion for users, especially if one accidentally adds/removes a ", 1".

Can we reconsider doing clamp(v, min, max) and rect(v, min, max) with the open start and closed end?